### ISSUE TITLE
Mustache rendering is getting old info or vars

### ISSUE THREAD

**Issue by adantoscano** [howdyai/botkit#1797]
### What are you trying to achieve or the steps to reproduce?
When I change info of a convo it doesn't change into nested object as cards

first time
````
dialog.setVar('name', db.name); // howdy
await dialog.gotoThread('send_web');

convo.addMessage('{{vars.name}}', 'send_web');  // howdy

  convo.addMessage(
    {
      channelData: {
        attachment: {
          type: 'template',
          payload: {
            template_type: 'generic',
            elements: [
              {
                title: 'Name',
                subtitle: 'Name',
                buttons: [
                  {
                    type: 'web_url',
                    url: '{{{vars.name}}}', //howdy
                    title: 'Name'
                  }
                ]
              }
            ]
          }
        }
      }
    },
    'send_web'
  );
````

second time
````
dialog.setVar('name', db.name); // botkit
await dialog.gotoThread('send_web');

convo.addMessage('{{vars.name}}', 'send_web');  // botkit

  convo.addMessage(
    {
      channelData: {
        attachment: {
          type: 'template',
          payload: {
            template_type: 'generic',
            elements: [
              {
                title: 'Name',
                subtitle: 'Name',
                buttons: [
                  {
                    type: 'web_url',
                    url: '{{{vars.name}}}', //howdy  !!!!! keeps the old value
                    title: 'Name'
                  }
                ]
              }
            ]
          }
        }
      }
    },
    'send_web'
  );
````

### What was the result you received?

Cards with old values

### What did you expect?

Actual values

### Context:

- Botkit version: 4.5.0
- Messaging Platform: Any
- Node version: 10.15
- Os: Linux
- Any other relevant information:

**Comment by benbrown** [comment]
Does your code indicate that you are calling convo.addMessage each time? Once the dialog is added to the controller, you should not make any further changes to the convo. Can you clarify?

**Comment by InnoraG** [comment]
Was facing similar issue, it was hard to debug being a variable copy issue.
Test my PR change https://github.com/howdyai/botkit/pull/1761 and confirm back if by any chance it works.
BTW just added similar sample.

**Comment by InnoraG** [comment]
Dear @adantoscano , have you found time to try the line var deepcopy to outgoing suggested by the PR ? What would be your feedback about the solution ?

**Comment by adantoscano** [comment]
@benbrown I do addMessage only once and setVar and goToThread into a before() that it gets info from my db and set this as a var to render.

Y think the error is related to @InnoraG says.

**Comment by benbrown** [comment]
I agree.

I will get this reviewed and merged into the next release.
